storm-core/src/clj/org/apache/storm/daemon/worker.clj

```diff
@@ -487,11 +486,12 @@
             )))))
 
 (defn register-callbacks [worker]
-  (log-message "Registering IConnectionCallbacks for " (:assignment-id worker) ":" (:port worker))
-  (msg-loader/register-callback (:transfer-local-fn worker)
-                                (:receiver worker)
-                                (:storm-conf worker)
-                                (worker-context worker)))
+  (let [transfer-local-fn (:transfer-local-fn worker)
+        ^IConnection socket (:receiver worker)]
+    (log-message "Registering IConnectionCallbacks for " (:assignment-id worker) ":" (:port worker))
+    (.registerRecv socket (DeserializingConnectionCallback. (:storm-conf worker)
```

Comment:
It wouldn't be that simple I suppose. There is a thread local KryoTupleDeserializer in the callback. I don't know how that will be implemented in clojure. I might end up porting more java code to clojure than other way around.